Quadriga/src/main/java/edu/asu/spring/quadriga/web/search/TextConceptSearchController.java

```diff
@@ -53,85 +52,83 @@
     private IJsonCreator jsonCreator;
 
     @RequestMapping(value = "search/texts")
-    public String search(@RequestParam(defaultValue = "") String conceptId, Model model) throws Exception {
+    public String search(@RequestParam(value = "conceptid", defaultValue = "") String[] conceptIds, Model model)
+            throws Exception {
+
+        List<String> conceptUriSearchList = new ArrayList<String>();
+        List<IConcept> concepts = new ArrayList<IConcept>();
+        List<ITextFile> texts = new ArrayList<ITextFile>();
+        List<String> handles = new ArrayList<String>();
+
+        for (String conceptUri : conceptIds) {
+            if (!conceptUri.isEmpty()) {
+                IConcept concept = cpCache.getConceptByUri(conceptUri);
+                if (concept != null) {
+                    concept.setId(conceptUri);
+                }
 
-        String conceptUri = conceptId;
+                concepts.add(concept);
+                conceptUriSearchList.add(conceptUri);
 
-        if (!conceptId.isEmpty()) {
-            IConcept concept = cpCache.getConceptByUri(conceptId);
-            
-            if (concept != null) {
-                concept.setId(conceptUri);
             }
-
-            if (conceptId.startsWith("http://")) {
-                int lastIdx = conceptId.lastIndexOf("/");
-                conceptId = conceptId.substring(lastIdx + 1);
+        }
+        String results = null;
+        if (conceptUriSearchList.size() == 1) {
+            String conceptId = null;
+            if (conceptUriSearchList.get(0).startsWith("http://")) {
+                int lastIdx = conceptUriSearchList.get(0).lastIndexOf("/");
+                conceptId = conceptUriSearchList.get(0).substring(lastIdx + 1);
             }
+            results = qStoreConnector.searchNodesByConcept(conceptId);
+        } else if (conceptUriSearchList.size() == 2) {
+            results = qStoreConnector.loadNetworkWithConceptsBelongingToSameStatements(conceptUriSearchList);
+        }
```

Comment:
if you adjust the query to make it take any number of concepts, then this here would be just line 84 and all the rest can be removed.